crates/meilisearch-types/src/keys.rs

```diff
@@ -195,206 +196,294 @@ fn parse_expiration_date(
-    #[serde(rename = "keys.get")]
-    #[deserr(rename = "keys.get")]
-    KeysGet,
-    #[serde(rename = "keys.update")]
-    #[deserr(rename = "keys.update")]
-    KeysUpdate,
-    #[serde(rename = "keys.delete")]
-    #[deserr(rename = "keys.delete")]
-    KeysDelete,
-    #[serde(rename = "experimental.get")]
-    #[deserr(rename = "experimental.get")]
-    ExperimentalFeaturesGet,
-    #[serde(rename = "experimental.update")]
-    #[deserr(rename = "experimental.update")]
-    ExperimentalFeaturesUpdate,
-    #[serde(rename = "network.get")]
-    #[deserr(rename = "network.get")]
-    NetworkGet,
-    #[serde(rename = "network.update")]
-    #[deserr(rename = "network.update")]
-    NetworkUpdate,
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, PartialOrd, Ord, ToSchema)]
+#[repr(transparent)]
+pub struct Action(u32);
+
+bitflags! {
+    // NOTE: For `Sequence` impl to work, the values of these must be in ascending order
+    impl Action: u32 {
+        const Search = 1;
+        // Documents
+        const DocumentsAdd = 1 << 1;
+        const DocumentsGet = 1 << 2;
+        const DocumentsDelete = 1 << 3;
+        const DocumentsAll = Self::DocumentsAdd.bits() | Self::DocumentsGet.bits() | Self::DocumentsDelete.bits();
+        // Indexes
+        const IndexesAdd = 1 << 4;
+        const IndexesGet = 1 << 5;
+        const IndexesUpdate = 1 << 6;
+        const IndexesDelete = 1 << 7;
+        const IndexesSwap = 1 << 8;
+        const IndexesAll = Self::IndexesAdd.bits() | Self::IndexesGet.bits() | Self::IndexesUpdate.bits() | Self::IndexesDelete.bits() | Self::IndexesSwap.bits();
+        // Tasks
+        const TasksCancel = 1 << 9;
+        const TasksDelete = 1 << 10;
+        const TasksGet = 1 << 11;
+        const TasksAll = Self::TasksCancel.bits() | Self::TasksDelete.bits() | Self::TasksGet.bits();
+        // Settings
+        const SettingsGet = 1 << 12;
+        const SettingsUpdate = 1 << 13;
+        const SettingsAll = Self::SettingsGet.bits() | Self::SettingsUpdate.bits();
+        // Stats
+        const StatsGet = 1 << 14;
+        const StatsAll = Self::StatsGet.bits();
+        // Metrics
+        const MetricsGet = 1 << 15;
+        const MetricsAll = Self::MetricsGet.bits();
+        // Dumps
+        const DumpsCreate = 1 << 16;
+        const DumpsAll = Self::DumpsCreate.bits();
+        // Snapshots
+        const SnapshotsCreate = 1 << 17;
+        const SnapshotsAll = Self::SnapshotsCreate.bits();
+        // Keys without an "all" version
+        const Version = 1 << 18;
+        const KeysAdd = 1 << 19;
+        const KeysGet = 1 << 20;
+        const KeysUpdate = 1 << 21;
+        const KeysDelete = 1 << 22;
+        // Experimental Features
+        const ExperimentalFeaturesGet = 1 << 23;
+        const ExperimentalFeaturesUpdate = 1 << 24;
+        // Network
+        const NetworkGet = 1 << 25;
+        const NetworkUpdate = 1 << 26;
+        // All
+        const All = 0xFFFFFFFF >> (32 - 1 - 26);
```

Comment:
Shouldn't we remove this variant and use [this `all` method](https://docs.rs/bitflags/latest/bitflags/trait.Flags.html#method.all), instead?

Reply:
Because I don't quite like having to track `26` by hand 😞

Reply:
It is not possible without changing the database